src/shared_utilities/SpaceVaryingInputs.jl

```diff
@@ -0,0 +1,133 @@
+# SpaceVaryingInputs.jl
+#
+# This module contains methods to process external data, regrid it onto the
+# model grid, and return the corresponding fields for use in the simulation.
+# This module only concerns with external data which varies in space,
+# and not time. For temporally varying input, we refer you to `TimeVaryingInputs.jl`.
+
+# All spatially varying parameter fields are assumed to fit into memory.
+
+# The planned parameter underlying arrays are:
+# - one-dimensional (values prescribed as a function of depth at a site),
+# - two-dimensional (values prescribed globally at each lat/lon),
+# - three-dimensional (values prescribed as a function of depth globally)
+# - analytic (functions of the coordinates of the space)
+
+module SpaceVaryingInputs
+using ClimaCore
+using DocStringExtensions
+import ..searchsortednearest
+import ..linear_interpolation
+using ClimaLand.FileReader
+export SpaceVaryingInput
+
+# Analytic case
+"""
+    SpaceVaryingInput(data_function::Function, space::S) 
+    where {S <: ClimaCore.Spaces.AbstractSpace}
+  
+Returns the parameter field to be used in the model; appropriate when
+a parameter is defined using a function of the coordinates of the space.
+
+Pass the ``data" as a function `data_function` which takes coordinates as arguments, 
+and  the ClimaCore space of the model simulation.
+"""
+function SpaceVaryingInput(
+    data_function::Function,
+    space::ClimaCore.Spaces.AbstractSpace,
+)
+    model_value = ClimaCore.Fields.zeros(space)
+    coords = ClimaCore.Fields.coordinate_field(space)
+    @. model_value = data_function(coords)
+    return model_value
+end
+
+# 1-D Case
+"""
+    function SpaceVaryingInput(
+        data_z::AbstractArray,
+        data_value::AbstractArray,
+        space::S,
+    ) where {S <: ClimaCore.Spaces.CenterFiniteDifferenceSpace}
+
+Interpolates the variable with values given in `data_value` as a function of
+depths `data_z` to the model grid defined implicitly by `space`. Returns a
```

Comment:
could you just explain that `data_z` is an array of the depths `data_value` is defined on? Also, maybe `data_value` should be called `data_values` since it's an array of multiple values